pkg/services/provisioning/datasources/config_reader_test.go

```diff
@@ -308,40 +267,41 @@ func validateDatasourceV1(t *testing.T, dsCfg *configs) {
 	require.Equal(t, ds.UID, "test_uid")
 }
 
-type fakeRepository struct {
+type mockOrgStore struct{ ExpectedOrg *models.Org }
+
+func (m *mockOrgStore) GetOrgById(c context.Context, cmd *models.GetOrgByIdQuery) error {
+	cmd.Result = m.ExpectedOrg
+	return nil
+}
+
+type spyStore struct {
 	inserted []*models.AddDataSourceCommand
 	deleted  []*models.DeleteDataSourceCommand
 	updated  []*models.UpdateDataSourceCommand
-
-	loadAll []*models.DataSource
+	items    []*models.DataSource
 }
 
-func mockDelete(ctx context.Context, cmd *models.DeleteDataSourceCommand) error {
-	fakeRepo.deleted = append(fakeRepo.deleted, cmd)
-	return nil
+func (s *spyStore) GetDataSource(ctx context.Context, query *models.GetDataSourceQuery) error {
```

Comment:
we probably want to put this mock somewhere common, i remember i needed something similar.

Reply:
Currently I'd vote for duplicating the mocks/spies so that we would see what parts could be reusable.
It all would depend on how we split the store and controllers. Ideally, for each widely used store and service there should be a subpackage with a more-or-less customisable test double. But that would also imply rewriting most of the tests. Thus, I prefer to just keeping notes about what store interfaces can be reused and make bigger decisions later.